File: android/src/main/java/com/instantpaycamera/PhotoCaptureConfigMetadata.kt
package com.instantpaycamera

enum class CameraFacing {
    FRONT,
    BACK
}

enum class CaptureQuality {
    LOW,
    MEDIUM,
    HIGH
}

enum class CameraFlash {
    AUTO,
    ON,
    OFF,
}

enum class TorchMode {
    ON,
    OFF
}

data class PhotoCaptureConfigMetadata(
    val quality: CaptureQuality,
    val flash: CameraFlash,
    val saveToGallery: Boolean,
    val maxWidth: Int?,
    val maxHeight: Int?,
    val base64ImageOutput: Boolean?,
    val compressBase64ImageOutput: Boolean?,
    val captureSound: Boolean?,
)

File: ios/PhotoCaptureConfigMetadata.swift
//
//  PhotoCaptureConfigMetadata.swift
//  InstantpayCamera
//
//  Created by Dhananjay kumar on 21/03/26.
//

import Foundation

enum CaptureQuality: String, Codable {
    case HIGH = "HIGH"
    case MEDIUM = "MEDIUM"
    case LOW = "LOW"
}

enum CameraFlash: String, Codable {
    case AUTO = "AUTO"
    case ON = "ON"
    case OFF = "OFF"
}

public struct PhotoCaptureConfigMetadata: Codable {
    var photoConfigurationStatus: Bool = false
    var quality: CaptureQuality = CaptureQuality.MEDIUM
    var flash: CameraFlash = CameraFlash.AUTO
    var saveToGallery: Bool = false
    var maxWidth: Int? = 0
    var maxHeight: Int? = 0
    var base64ImageOutput: Bool? = false
    var compressBase64ImageOutput: Bool? = false
    var captureSound: Bool? = false
    
}
